serverless.yml

```diff
@@ -24,10 +24,55 @@ provider:
       - "sqs:SendMessage"
      Resource:
       Fn::GetAtt: [ SQSQueue, Arn ]
+   - Effect: Allow
+     Action:
+      - "sqs:SendMessage"
+     Resource:
+      - "Fn::GetAtt":
+        - ReceiverDeadLetterQueue
+        - Arn
   environment:
     HTTPOBS_API_URL: 'https://http-observatory.security.mozilla.org/api/v1'
+    SQS_URL:
+      Ref: SQSQueue
+
+# Adding some packaging information here to clean up
+# and to include the nmap static library
+package:
+  include:
+    - bin/*
+  exclude:
+    - .venv/**
+    - .git/**
+    - __pycache__/**
+    - node_modules/**
 
 functions:
+  onDemandPortScan:
+    handler: handler.addPortScanToQueue
+    events:
+      - http:
+          path: ondemand/portscan
+          method: PUT
+          cors: true
+  cronPortScan:
+    handler: handler.runScheduledPortScan
+    # Giving 5 minutes to run the TCP scan before it times outs
+    timeout: 300
+    events:
+      # Invoke Lambda function once a day
+      # We can fine grain it with cron() later
+      - schedule: 
+          rate: rate(1 day)
+          # Not enabling this by default
+          enabled: false
+  onDemandObservatoryScan:
+    handler: handler.addObservatoryScanToQueue
```

Comment:
As I mentioned above, I don't think we're restricted to binding this into a single handler.

For instance, this could be "observatory.queue" and simply import helpers that would be common logic that we need from handler.

Reply:
Sounds good, I will have to experiment. Last time I tried something like this I couldn't get it to work on serverless, but it could certainly be my (lack of) serverless knowledge...

Reply:
I think this is probably better for for another branch, it's a fair bit of re-write of the code.
@claudijd you can give it a go if you want.

Reply:
ok, after we land